util/a2e/atoe_utils.c

```diff
@@ -545,22 +553,25 @@ atoe_vsnprintf(char *str, size_t count, const char *fmt, va_list args)
 					/* ibm@8665
 					 * Add floating point support for dbl2str & flt2str
 					 */
-					char *b;
-					int len;
-
-					b = a2e((char *)pattern, fmt - pattern);
+					char *tempPattern = a2e((char *)pattern, fmt - pattern);
+					size_t freeCap = (this.end > this.buffer) ? (this.end - this.buffer) : 0;
+					double argument  = va_arg(args, double);
+					int requiredSpace = 0;
 
 					/* Extract a double from args, this works for both doubles
 					 * and floats,
 					 * NB if we use float for a single precision floating
 					 * point number the result is wrong.
 					 */
-					len = sprintf(this.buffer, b, va_arg(args, double));
-					free(b);
-					b = e2a_string(this.buffer);
-					strcpy(this.buffer, b);
-					free(b);
-					this.buffer += len;
+					requiredSpace = snprintf(this.buffer, freeCap, tempPattern, argument);
+					free(tempPattern);
+					CheckRet(requiredSpace);
+					if ((freeCap > 0) && (requiredSpace > 0)) {
+						tempPattern = e2a_string(this.buffer);
```

Comment:
The name `tempPattern` doesn't match the use here. Either use a more general name or introduce a new declaration here.

Reply:
Thanks @keithc-ca for your review. Let me know if my changes in the last commit satisfies your expectation so I can squash my commits.